timdex_dataset_api/record.py

```diff
@@ -1,12 +1,15 @@
 """timdex_dataset_api/record.py"""
 
-import datetime
-from dataclasses import asdict, dataclass
+from datetime import UTC, datetime
 
-from timdex_dataset_api.exceptions import InvalidDatasetRecordError
+from attrs import asdict, define, field
 
 
-@dataclass
+def strict_date_parse(date_string: str) -> datetime:
+    return datetime.strptime(date_string, "%Y-%m-%d").astimezone(UTC)
```

Comment:
Initially thought we could use `dateutil.parser` as a converter, but it seems that the parser will make some assumptions for _almost_ correctly formatted date strings. For example: 

```
dateutil.parse("-02-01") -> datetime.datetime(2024, 2, 1)
```

It will interpret the provided value as a **relative date** and fill in the missing gaps with default values (i.e., the current year).

Reply:
Nice catch.  That could have been quietly problematic if not noticed!